allennlp/modules/attention/legacy_attention.py

```diff
@@ -0,0 +1,39 @@
+"""
+An *attention* module that computes the similarity between
+an input vector and the rows of a matrix.
+"""
+
+import torch
+
+from overrides import overrides
+from allennlp.common import Params
+from allennlp.modules.attention.attention import Attention
+from allennlp.modules.similarity_functions import DotProductSimilarity, SimilarityFunction
+
+
+@Attention.register("legacy")
+class LegacyAttention(Attention):
+
```

Comment:
Again here, remove the module docstring and add a class docstring.  This one should say something about using a configurable `SimilarityFunction` to compute the attention, but in a way that uses more memory than it needs to.  This is deprecated in favor of more efficient `Attention` implementations that don't use `SimilarityFunctions`.